omni-relayer/src/main.rs

```diff
@@ -0,0 +1,22 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+
+mod defaults;
+mod startup;
+mod utils;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    pretty_env_logger::init();
+
+    let near_signer = startup::near::create_signer()?;
+    let connector = Arc::new(startup::build_connector(&near_signer)?);
+
+    tokio::spawn(startup::near::start_indexer(near_signer, connector.clone()));
```

Comment:
Let's add more async workers.
1- near_sing_transfer_worker
2- eth_fin_transfer_worker
3- near_claim_fee_worker
4- eth_track_events_worker
5- near_track_events_order

The events, transaction hashes, and nonces should be stored in the Redis DB.
Also, additional workers could be added for transaction status tracking to avoid blocking the primary workers.

Reply:
> Let's add more async workers.
> 1- near_sing_transfer_worker
> 2- eth_fin_transfer_worker
> 3- near_claim_fee_worker
> 4- eth_track_events_worker
> 5- near_track_events_order

Can you please take a look at the new version? The first thought idea that came to my mind was to use simple channels, but:
a) it made code more complicated without any real reason to do so
b) Now as we use channels we can't really call two methods at the same time

I guess you meant something different, can you please elaborate a bit then? For example, in previous approach we had `sign_transfer` async call directly from `near_track_events`. If we want separate these workers I don't see other approach than using channels, but as I've mentioned above it only made things worse in my opinion

Reply:
Please take a look at this implementation https://github.com/Near-One/fast-bridge-service/blob/master/src/main.rs.
The workers are synced just by reading from the redis-db, but we optimize it with observer pattern.
This is needed to guarantee data and process integrity; the service should correctly resume if it was killed or restarted.

Reply:
The code is far from being the cleanest, but the core logic works, and everything is synchronized through Redis. Could you please take a look to see if this is the desired implementation?